relayer/relays/beacon/header/syncer/syncer.go

```diff
@@ -163,12 +188,104 @@ func (s *Syncer) GetSyncCommitteePeriodUpdate(from uint64) (SyncCommitteePeriodU
 	return syncCommitteePeriodUpdate, err
 }
 
-func (s *Syncer) GetFinalizedUpdate() (FinalizedHeaderUpdate, common.Hash, error) {
+func (s *Syncer) GetBlockRoots(ctx context.Context, stateRoot string) (types.H256, []types.H256, error) {
+	networkConfig := params.MainnetConfig()
+
+	if s.network.IsGoerli() {
+		networkConfig = params.PraterConfig()
+	}
+	if s.network.IsLocal() {
+		return types.H256{}, nil, ErrLocalNetBeaconStateNotSupported
+	}
+
+	err := s.Client.DownloadBeaconState(stateRoot)
+	switch {
+	case errors.Is(err, ErrNotFound):
+		return types.H256{}, nil, ErrBeaconStateAvailableYet
+	case err != nil:
+		return types.H256{}, nil, err
+	}
+
+	const fileName = "beacon_state.ssz"
+
+	defer func() {
+		_ = os.Remove(fileName)
+	}()
+
+	data, err := os.ReadFile(fileName)
+	if err != nil {
+		return types.H256{}, nil, fmt.Errorf("find beacon state file: %w", err)
+	}
+
+	defer func() {
+		_ = os.Remove(fileName)
+	}()
+
+	params.OverrideBeaconConfig(networkConfig)
+
+	cf, err := detect.FromState(data)
+	if err != nil {
+		return types.H256{}, nil, fmt.Errorf("detect network from beacon state: %w", err)
+	}
+
+	_, ok := params.BeaconConfig().ForkVersionSchedule[cf.Version]
+	if !ok {
+		return types.H256{}, nil, fmt.Errorf("config mismatch, beacon node configured to connect to %s, detected state is for %s", params.BeaconConfig().ConfigName, cf.Config.ConfigName)
+	}
+
+	state, err := cf.UnmarshalBeaconState(data)
+	if err != nil {
+		return types.H256{}, nil, fmt.Errorf("unmarshal beacon state using prysm: %w", err)
+	}
+
+	currentBlockHash, err := state.LatestBlockHeader().HashTreeRoot()
+
+	log.WithField("currentFinalizedHeader", s.currentFinalizedHeader).Info("setting current finalized header")
+
+	blockRootsHash, err := stateutil.ArraysRoot(state.BlockRoots(), uint64(len(state.BlockRoots())))
```

Comment:
@claravanstaden  notice that we did some [customization](https://github.com/Snowfork/prysm/commit/4aec9d4e72a1f946dcd7088ce8018328f47518b1) in `prysm` to add proof for blockRoots which we will use later for the ancestry execution block verification, right? 
  But the original code I checked [here](https://github.com/Snowfork/prysm/blob/4aec9d4e72a1f946dcd7088ce8018328f47518b1/beacon-chain/state/state-native/hasher.go#L72) to calculate the leaf is based on `SLOTS_PER_HISTORICAL_ROOT` while we use the current length here.
  so is that correct or I misunderstand something?

Reply:
Hey @yrong! Nice catch! You are right, it should be `SLOTS_PER_HISTORICAL_ROOT`. However, I am going to likely going remove the Prysm dependency and rather use https://github.com/ferranbt/fastssz. I am busy fixing this issue at the moment: https://github.com/ferranbt/fastssz/issues/119.

Reply:
I've removed this since using Prysm is no longer necessary.